From 6b7f2a3fa527e7f8d7bddd20dc7b717b273cf784 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:04 +0800
Subject: [PATCH 1/4] GSI Android 17 compatibility:
 0001-Add-a-vendor800-command-to-be-able-to-call-vendor-sp.patch

Change-Id: I91d9c2dd93ba96cdccd0da1338a4ea19a91821a1
---
 .../composer-command-buffer/2.1/ComposerCommandBuffer.h     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/graphics/composer/2.1/utils/command-buffer/include/composer-command-buffer/2.1/ComposerCommandBuffer.h b/graphics/composer/2.1/utils/command-buffer/include/composer-command-buffer/2.1/ComposerCommandBuffer.h
index 9ce6eed..aff5b8d 100644
--- a/graphics/composer/2.1/utils/command-buffer/include/composer-command-buffer/2.1/ComposerCommandBuffer.h
+++ b/graphics/composer/2.1/utils/command-buffer/include/composer-command-buffer/2.1/ComposerCommandBuffer.h
@@ -402,6 +402,12 @@ class CommandWriterBase {
         endCommand();
     }
 
+    void vendor800_1(uint32_t val) {
+        beginCommand(0x800 << 16, 1);
+        write(val);
+        endCommand();
+    }
+
    protected:
      template <typename T>
      void beginCommand(T command, uint16_t length) {
